# просим пользователя ввести 2 числа чтобы поделить одно на другое
# a = int(input("Введите делимое: "))
# b = int(input("введите делитель: "))
# try:
#     c = a / b
#     print(c)
# except:
#     print("Возникла ошибка!")

# давайте добавим словарь
# d = {
#     "key_1": "value_1",
#     "key_2": "value_2"
# }
#
# a = 1
# b = 0
#
# try:
#     print(d["key_2"])
#     c = a / b
# except (ZeroDivisionError, KeyError) as err:  # перехватываем определённые типы исключений
#     print("Возникла ошибка!", type(err), err)

# очень простая функция, которая возвращает результат деления числа a на число b
def divisor(a, b):
    print("Работает функция!")
    return a / b

first_num = 1
second_num = 2

# позиционные аргументы
# print(divisor(1, 2))

# передача аргументов по ключу
divisor(b=2, a=10)

# почему информация не выводится?? потому что результат возвращается для последующей работы, но мы не сказали печатать
# его на экран:
print(divisor(1, 2))
